sdk/core/core-amqp/src/auth/sas.ts

```diff
@@ -78,7 +84,98 @@ export class SharedKeyCredential {
    * @param {string} connectionString - The EventHub/ServiceBus connection string
    */
   static fromConnectionString(connectionString: string): SharedKeyCredential {
-    const parsed = parseConnectionString<ServiceBusConnectionStringModel>(connectionString);
-    return new SharedKeyCredential(parsed.SharedAccessKeyName, parsed.SharedAccessKey);
+    const parsed = parseConnectionString<
+      ServiceBusConnectionStringModel & { SharedAccessSignature: string }
+    >(connectionString);
+
+    if (parsed.SharedAccessSignature == null) {
+      return new SharedKeyCredential(parsed.SharedAccessKeyName, parsed.SharedAccessKey);
+    } else {
+      return new SharedAccessSignatureCredential(parsed.SharedAccessSignature);
+    }
+  }
+}
+
+/**
+ * A credential that takes a SharedAccessSignature of two forms:
+ * `SharedAccessSignature sr=<resource>&sig=<signature>&se=<expiry>&skn=<keyname>`
+ * or
+ * `sr=<resource>&sig=<signature>&se=<expiry>&skn=<keyname>`
+ *
+ * @internal
+ * @ignore
+ */
+export class SharedAccessSignatureCredential extends SharedKeyCredential {
```

Comment:
What does extending from SharedKeyCredential give us other than avoiding another branch in the `if/else` when we want to call `getToken()` in SB/EH?

Reply:
That's the benefit but it's a big one since it avoids lock in on our part since core-amqp is published.

Reply:
I've updated things so there's much less code but keeping this part of having `SharedAccessSignatureCredential`

I've left this part in since it keeps the API-level changes pretty small.